modules/utxo-lib/src/bitgo/signature.ts

```diff
@@ -80,21 +79,25 @@ export interface ParsedSignatureScriptP2PKH extends ParsedSignatureScript {
 export interface ParsedSignatureScript2Of3 extends ParsedSignatureScript {
   scriptType: 'p2sh' | 'p2shP2wsh' | 'p2wsh';
   inputClassification: 'scripthash' | 'witnessscripthash';
+  publicKeys: [Buffer, Buffer, Buffer];
   signatures:
     | [Buffer, Buffer] // fully-signed transactions with signatures
     /* Partially signed transactions with placeholder signatures.
        For p2sh, the placeholder is OP_0 (number 0) */
     | [Buffer | 0, Buffer | 0, Buffer | 0];
-  publicKeys: [Buffer, Buffer, Buffer];
   pubScript: Buffer;
 }
 
 export interface ParsedSignatureScriptTaproot extends ParsedSignatureScript {
   scriptType: 'p2tr';
   inputClassification: 'taproot';
-  // P2TR tapscript spends are for 2-of-2 multisig scripts
-  signatures: [Buffer, Buffer]; // missing signature is encoded as empty buffer
-  publicKeys: [Buffer, Buffer];
+  // P2TR tapscript spends are for keypath spends or 2-of-2 multisig scripts
+  // A single signature indicates a keypath spend.
+  // Two signatures indicate a scriptPath spend.
+  publicKeys: [Buffer] | [Buffer, Buffer];
+  signatures: [Buffer] | [Buffer, Buffer];
+  // For scriptpath signatures, this indicates level. For keypath signatures this is undefined.
+  scriptPathLevel: number | undefined;
   pubScript: Buffer;
   controlBlock: Buffer;
```

Comment:
Thinking about this again, `controlBlock` ought to be as optional as it will be undefined for keypath spends.

Reply:
will change